dockstore-webservice/src/main/java/io/dockstore/webservice/resources/AbstractWorkflowResource.java

```diff
@@ -170,6 +171,12 @@ protected void updateDBWorkflowWithSourceControlWorkflow(Workflow workflow, Work
         for (WorkflowVersion version : newWorkflow.getWorkflowVersions()) {
             // skip frozen versions
             WorkflowVersion workflowVersionFromDB = existingVersionMap.get(version.getName());
+
+            // Skip these since they have not changed
+            if (Objects.equals(SKIP_COMMIT_ID, version.getCommitID())) {
```

Comment:
I would find this easier to read if you had a filter on line 170/171.

Alternatively if you don't want to do the filter, move it before line 172/173, because you don't need to do the map lookup -- suggesting it not as a micro-optimization, which it would be, but more because I was trying to understand this code and wondering why a map lookup was being done before this if condition.